packages/web3-utils/src/utils.js

```diff
@@ -84,13 +84,11 @@ var toTwosComplement = function (number) {
  */
 var isAddress = function (address) {
     // check if it has the basic requirements of an address
-    if (!/^(0x)?[0-9a-f]{40}$/i.test(address)) {
-        return false;
-        // If it's ALL lowercase or ALL upppercase
-    } else if (/^(0x|0X)?[0-9a-f]{40}$/.test(address) || /^(0x|0X)?[0-9A-F]{40}$/.test(address)) {
+    //case insensitive regexp
+    if (/^(0x)?[0-9a-f]{40}$/i.test(address)) {
```

Comment:
The earlier logic was only accepting either all upper case or all lower case. This change will allow mixed case addresses. I don't think it's appropriate to change.

Reply:
@nazarhussain checksum addresses are mixed case so its valid